sdk/src/main/java/com/kin/ecosystem/settings/presenter/SettingsPresenter.java

```diff
@@ -0,0 +1,172 @@
+package com.kin.ecosystem.settings.presenter;
+
+import static com.kin.ecosystem.settings.view.ISettingsView.BLUE;
+import static com.kin.ecosystem.settings.view.ISettingsView.GRAY;
+import static com.kin.ecosystem.settings.view.ISettingsView.ITEM_BACKUP;
+
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import com.kin.ecosystem.backup.BackupCallback;
+import com.kin.ecosystem.backup.BackupEvents;
+import com.kin.ecosystem.backup.BackupManager;
+import com.kin.ecosystem.backup.RestoreCallback;
+import com.kin.ecosystem.backup.RestoreEvents;
+import com.kin.ecosystem.base.BasePresenter;
+import com.kin.ecosystem.common.Observer;
+import com.kin.ecosystem.common.model.Balance;
+import com.kin.ecosystem.core.bi.EventLogger;
+import com.kin.ecosystem.core.data.blockchain.BlockchainSource;
+import com.kin.ecosystem.core.data.settings.SettingsDataSource;
+import com.kin.ecosystem.settings.view.ISettingsView;
+import com.kin.ecosystem.settings.view.ISettingsView.IconColor;
+import com.kin.ecosystem.settings.view.ISettingsView.Item;
+import java.math.BigDecimal;
+
+public class SettingsPresenter extends BasePresenter<ISettingsView> implements ISettingsPresenter {
+
+	private final BackupManager backupManager;
+	private final SettingsDataSource settingsDataSource;
+	private final BlockchainSource blockchainSource;
+	private final EventLogger eventLogger;
+
+	private Observer<Balance> balanceObserver;
+	private Balance currentBalance;
+
+	public SettingsPresenter(@NonNull final ISettingsView view, @NonNull final SettingsDataSource settingsDataSource,
+		@NonNull final BlockchainSource blockchainSource, @NonNull final BackupManager backupManager,
+		@NonNull final EventLogger eventLogger) {
+		this.view = view;
+		this.backupManager = backupManager;
+		this.settingsDataSource = settingsDataSource;
+		this.blockchainSource = blockchainSource;
+		this.eventLogger = eventLogger;
+		this.currentBalance = blockchainSource.getBalance();
+		registerToCallbacks();
+		registerToEvents();
+
+		this.view.attachPresenter(this);
+	}
+
+	@Override
+	public void onAttach(ISettingsView view) {
+		super.onAttach(view);
+		addBalanceObserver();
+	}
+
+	@Override
+	public void onDetach() {
+		super.onDetach();
+		removeBalanceObserver();
+	}
+
+	private void addBalanceObserver() {
+		balanceObserver = new Observer<Balance>() {
+			@Override
+			public void onChanged(Balance value) {
+				currentBalance = value;
+				updateBalanceSettingsIcon();
+			}
+		};
+		blockchainSource.addBalanceObserver(balanceObserver);
+	}
+
+	private void updateBalanceSettingsIcon() {
+		if (!settingsDataSource.isBackedUp()) {
+			changeIconColor(ITEM_BACKUP, GRAY);
```

Comment:
this is duplicate code - it might be good idea to have a presenter that only handle the setting button and used on both activities